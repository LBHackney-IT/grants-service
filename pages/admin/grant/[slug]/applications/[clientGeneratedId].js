import { useRouter } from 'next/router';

import { redirectIfNotAuth } from '../../../../../utils/auth';
import ApplicationView from '../../../../../components/ApplicationView/ApplicationView';
import { getGrantBySlug } from '../../../../../grants/grants';

const ApplicationViewPage = () => {
  const { query } = useRouter();
  const { clientGeneratedId, ...queryParameters } = query;

  const slug = Array.isArray(queryParameters.slug)
    ? queryParameters.slug[0]
    : queryParameters.slug;

  const grant = getGrantBySlug(slug);

  return (
    <>
      <div className="govuk-breadcrumbs">
        <ol className="govuk-breadcrumbs__list">
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" href="/admin">
              Manage grant applications
            </a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" href="/admin">
              {grant.name}
            </a>
          </li>
          <li className="govuk-breadcrumbs__list-item">Application</li>
        </ol>
      </div>

      <ApplicationView applicationId={clientGeneratedId} />
    </>
  );
};

export const getServerSideProps = redirectIfNotAuth;

export default ApplicationViewPage;
