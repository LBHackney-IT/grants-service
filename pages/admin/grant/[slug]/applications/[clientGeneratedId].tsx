import { useRouter } from 'next/router';
import React from 'react';

import { redirectIfNotAuth } from '../../../../../utils/auth';
import ApplicationView from '../../../../../components/ApplicationView/ApplicationView';
import { getGrantBySlug } from '../../../../../grants/grants';

const ApplicationViewPage: React.FC = () => {
  const { query } = useRouter();

  const slug = Array.isArray(query.slug) ? query.slug[0] : query.slug;

  const grant = getGrantBySlug(slug);

  const clientGeneratedId = Array.isArray(query.clientGeneratedId)
    ? query.clientGeneratedId[0]
    : query.clientGeneratedId;

  return (
    <>
      <div className="govuk-breadcrumbs">
        <ol className="govuk-breadcrumbs__list">
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" href="/admin">
              Manage grant applications
            </a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" href="/admin">
              {grant.name}
            </a>
          </li>
          <li className="govuk-breadcrumbs__list-item">Application</li>
        </ol>
      </div>

      <ApplicationView applicationId={clientGeneratedId} />
    </>
  );
};

export const getServerSideProps = redirectIfNotAuth;

export default ApplicationViewPage;
