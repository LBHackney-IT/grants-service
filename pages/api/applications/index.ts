import * as HttpStatus from 'http-status-codes';
import { customAlphabet } from 'nanoid';
import AppContainer from '../../../containers/AppContainer';
import { getUserStringFromCookie } from '../../../utils/auth';
import {
  PAGE_MUST_BE_AT_LEAST_ONE,
  PAGINATED_PAST_END,
} from '../../../lib/usecases/listApplications';
import uploadApplication from '../../../lib/usecases/uploadApplication';
import isValidApplication from '../../../lib/usecases/validators';
import sendConfirmationEmail from '../../../lib/usecases/sendConfirmationEmail';

export default async (req, res) => {
  const rightNow = new Date();
  const lastSubmission = new Date(process.env.EXPIRATION_DATE);

  switch (req.method) {
    case 'GET':
      try {
        const container = AppContainer.getInstance();
        const listApplications = container.getListApplications();
        res.setHeader('Content-Type', 'application/json');

        if (!req.query.grantType) {
          throw new Error("Missing 'grantType' query parameter");
        }

        const grantType = req.query.grantType as string;

        const currentPage =
          req.query && req.query.page
            ? parseInt(req.query.page, 10)
            : undefined;
        const pageSize =
          req.query && req.query.pageSize
            ? parseInt(req.query.pageSize, 10)
            : undefined;
        const sort = req.query && req.query.sort ? req.query.sort : undefined;
        const status =
          req.query && req.query.status ? req.query.status : undefined;
        const searchTerm =
          req.query && req.query.searchTerm ? req.query.searchTerm : undefined;

        const businessCategory =
          req.query && req.query.businessCategory
            ? req.query.businessCategory
            : undefined;

        const businessSize =
          req.query && req.query.businessSize
            ? req.query.businessSize
            : undefined;

        const businessPremises =
          req.query && req.query.businessPremises
            ? req.query.businessPremises
            : undefined;

        const date = req.query && req.query.date ? req.query.date : undefined;

        const grantOfficer =
          req.query && req.query.grantOfficer
            ? req.query.grantOfficer
            : undefined;

        const clientGeneratedId =
          req.query && req.query.applicationId
            ? req.query.applicationId
            : undefined;

<<<<<<< HEAD
        const listApplicationsResponse = await listApplications({
=======
        let listApplicationsResponse = await listApplications({
>>>>>>> feat: split the admin panel to support multiple grants
          grantType,
          currentPage,
          pageSize,
          sort,
          status,
          businessCategory,
          grantOfficer,
          clientGeneratedId,
          businessSize,
          businessPremises,
          date,
          searchTerm,
        });
        if (
          [PAGE_MUST_BE_AT_LEAST_ONE, PAGINATED_PAST_END].includes(
            listApplicationsResponse.error
          )
        ) {
          res.statusCode = HttpStatus.BAD_REQUEST;
        } else {
          res.statusCode = HttpStatus.OK;
        }
        res.end(JSON.stringify(listApplicationsResponse));
      } catch (error) {
        console.log('Application list error:', error);
        res.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
        res.end(JSON.stringify('Unable to list applications'));
      }
      break;

    case 'POST':
      if (rightNow >= lastSubmission) {
        res.statusCode = HttpStatus.FORBIDDEN;
        res.setHeader('Content-Type', 'application/json');
        res.end(JSON.stringify({ error: 'Applications have closed.' }));
        break;
      }

      try {
        const nanoid = customAlphabet(
          '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz0123456789',
          21
        );
        const clientGeneratedId = nanoid();
        const validApplication = await isValidApplication(req.body);
        await uploadApplication({ ...validApplication, clientGeneratedId });
        await sendConfirmationEmail(
          clientGeneratedId,
          req.body.contact.emailAddress
        );
        res.statusCode = HttpStatus.CREATED;
        res.setHeader('Content-Type', 'application/json');
        res.end(JSON.stringify(clientGeneratedId));
      } catch (error) {
        console.log('Application submission error:', error, 'request:', req);
        // Todo: We should 400 on invalid application and 500 on Internal Server Error
        res.statusCode = HttpStatus.BAD_REQUEST;
        res.end(JSON.stringify(error.message));
      }
      break;

    case 'PATCH':
      try {
        const container = AppContainer.getInstance();
        const patchApplications = container.getPatchApplications();
        res.statusCode = HttpStatus.OK;
        res.setHeader('Content-Type', 'text/csv; charset=utf-8');
        res.setHeader('Content-Disposition', 'filename=export.csv');
        const patchResponse = await patchApplications({
          author: getUserStringFromCookie(req.headers.cookie),
          round: req.body.round,
        });
        res.end(patchResponse.csvString);
      } catch (error) {
        console.log('Applications patch error:', error, 'request:', req);
        res.statusCode = HttpStatus.BAD_REQUEST;
        res.end(JSON.stringify(error.message));
      }
      break;

    default:
      res.statusCode = HttpStatus.BAD_REQUEST;
      res.end(JSON.stringify('Invalid request method'));
  }
};
