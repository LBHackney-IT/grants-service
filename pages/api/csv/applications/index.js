import * as HttpStatus from 'http-status-codes';
import AppContainer from '../../../../containers/AppContainer';

export default async (req, res) => {
  switch (req.method) {
    case 'GET':
      try {
        const grantType = req.query.grantType;
        if (!grantType) {
          throw new Error("Missing 'grantType' parameter");
        }

        const container = AppContainer.getInstance();
        const listApplicationsCSV = container.getListApplicationsCSV();
        res.statusCode = HttpStatus.OK;
        res.setHeader('Content-Type', 'text/csv; charset=utf-8');
        res.setHeader('Content-Disposition', 'filename=data.csv');
        const csvResult = await listApplicationsCSV({
          grantType,
        });
        res.end(csvResult.csvString);
      } catch (error) {
        console.log('Applications CSV error:', error);
        res.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
        res.end(JSON.stringify('Unable to generate an applications CSV'));
      }
      break;

    default:
      res.statusCode = HttpStatus.BAD_REQUEST;
      res.end(JSON.stringify('Invalid request method'));
  }
};
