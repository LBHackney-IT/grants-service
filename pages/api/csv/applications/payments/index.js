import * as HttpStatus from 'http-status-codes';
import AppContainer from '../../../../../containers/AppContainer';
import {
  getUserStringFromCookie,
  getUserFromCookie,
} from '../../../../../utils/auth';

export default async (req, res) => {
  switch (req.method) {
    case 'GET':
      try {
        const grantType = req.query.grantType;
        if (!grantType) {
          throw new Error("Missing 'grantType' parameter");
        }

        const user = getUserFromCookie(req.headers.cookie);
        if (!user.groups.includes(process.env.CSV_DOWNLOAD_GROUP)) {
          res.statusCode = HttpStatus.FORBIDDEN;
          res.end(JSON.stringify('Access forbidden'));
        }

        const container = AppContainer.getInstance();
        const listApplicationsCSV = container.getPatchApplications();
        res.statusCode = HttpStatus.OK;
        res.setHeader('Content-Type', 'text/csv; charset=utf-8');
        res.setHeader('Content-Disposition', 'filename=payments.csv');
        const csvResult = await listApplicationsCSV({
          grantType,
          author: getUserStringFromCookie(req.headers.cookie),
        });
        res.end(csvResult.csvString);
      } catch (error) {
        console.log('Patch Applications CSV error:', error);
        res.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
        res.end(
          JSON.stringify('Unable to generate a payment applications CSV')
        );
      }
      break;

    default:
      res.statusCode = HttpStatus.BAD_REQUEST;
      res.end(JSON.stringify('Invalid request method'));
  }
};
