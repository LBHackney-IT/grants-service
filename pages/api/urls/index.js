import * as HttpStatus from 'http-status-codes';
import getSecureUploadUrl from '../../../lib/usecases/getSecureUploadUrl';
import { fileTypeFromFile } from 'file-type';
import { IncomingForm } from 'formidable';

const allowedMimes = [
  'application/pdf',
  'image/jpeg',
  'image/png',
  'application/msword',
  'application/vnd.openxmlformats',
  'officedocument.wordprocessingml.document',
  'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
  'application/vnd.ms-excel',
];

export const config = {
  api: {
    bodyParser: false,
  },
};

export default async (req, res) => {
  try {
    const data = await new Promise((resolve, reject) => {
      const form = new IncomingForm();

      form.parse(req, (err, fields, files) => {
        if (err) return reject(err);
        resolve({ fields, files });
      });
    });

    const { clientGeneratedId, fileName } = data.fields;
    const fileType = await fileTypeFromFile(data?.files?.file.filepath);

    if (fileType && !allowedMimes.includes(fileType.mime))
      throw new Error('Disallowed mime type');

    const { documentId, fields, url } = await getSecureUploadUrl(
      clientGeneratedId,
      fileName
    );

    const fileKey = `${clientGeneratedId}/${documentId}/${fileName}`;
    res.statusCode = HttpStatus.OK;
    res.setHeader('Content-Type', 'application/json');
    res.end(
      JSON.stringify({
        fileKey,
        url,
        fields: {
          key: fileKey,
          ...fields,
          'X-Amz-Server-Side-Encryption': 'AES256',
          'X-Amz-Meta-Description': clientGeneratedId,
        },
      })
    );
  } catch (error) {
    console.log(error);
    res.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
    res.end();
  }
};
