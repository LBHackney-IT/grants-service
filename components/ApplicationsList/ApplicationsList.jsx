import { useState, useEffect, useMemo, useCallback } from 'react';
import Router, { useRouter } from 'next/router';

import Table from '../Table/Table';
import ErrorMessage from '../ErrorMessage/ErrorMessage';
import { BasicSelect, TextInput } from '../Form';
import {
  fetchApplications,
  patchApplications,
} from '../../utils/api/applications';

import {
  APPLICATION_STATE,
  BUSINESS_CATEGORIES,
  BUSINESS_SIZE,
  DATES,
  TYPE_OF_BUSINESS,
} from '../../lib/dbMapping';
import { fetchGrantOfficers } from '../../utils/api/grantOfficers';

const ApplicationsList = ({
  grantType,
  page,
  pageSize,
  sort,
  status,
  grantOfficer,
  businessCategory,
  businessSize,
  businessPremises,
  applicationId,
  searchTerm,
}) => {
  const columns = useMemo(
    () => [
      {
        Header: 'Business Name',
        accessor: 'businessName',
      },
      {
        Header: 'Application ID',
        accessor: 'clientGeneratedId',
        disableSortBy: true,
      },
      {
        Header: 'Submission',
        accessor: 'applicationDate',
        Cell: ({ value }) => new Date(value).toLocaleString(),
      },
      {
        Header: 'Status',
        accessor: 'status',
        disableSortBy: true,
      },
    ],
    []
  );
  const [filters, setFilters] = useState({
    status,
    grantOfficer,
    applicationId,
    businessCategory,
    businessSize,
    businessPremises,
    searchTerm,
  });
  const [error, setError] = useState();
  const [searchTermInput, setSearchTermInput] = useState(searchTerm);
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [pageCount, setPageCount] = useState(0);
  const [officers, setOfficers] = useState([]);
  const [totalRecords, setTotalRecords] = useState(0);

  useEffect(() => {
    const fetchOfficers = async () => {
      try {
        setError(null);
        const data = await fetchGrantOfficers();
        setOfficers(data.grantOfficers.map(({ identifier }) => identifier));
      } catch (e) {
        setError(e.response.data);
        setOfficers(null);
      }
    };

    fetchOfficers();
  }, []);

  useEffect(() => {
    fetchData(filters);
  }, [filters]);

  const setValues = useCallback((state) =>
    setFilters({ ...filters, ...state })
  );

  const fetchData = useCallback(
    async ({ pageSize, pageIndex, sortBy, ...otherFilters }) => {
      if (isNaN(pageSize)) {
        return;
      }

      setLoading(true);

      const query = {
        grantType,
        page: pageIndex + 1,
        pageSize,
        sort: sortBy && `${sortBy.desc ? '-' : '+'}${sortBy.id}`,
        ...otherFilters,
      };

      await Router.push(
        `/admin/grant/${grantType}`,
        {
          pathname: `/admin/grant/${grantType}`,
          query,
        },
        { shallow: true }
      );

      try {
        setError(null);
        const { applications, pagination } = await fetchApplications(query);
        setData(applications);
        setPageCount(pagination.totalPages);
        setTotalRecords(pagination.totalRecords);
        setLoading(false);
      } catch (e) {
        if (e.response.status === 400) {
          setValues({ pageIndex: 0, pageSize, sortBy, ...otherFilters });
          return;
        }

        if (e.response.status === 404) {
          setError(e.message);
          return;
        }

        setError(e.response.data);
      }
    },
    []
  );

  const handleSearchTermChange = (event) => {
    setSearchTermInput(event.target.value);
  };

  const handleSearch = (e) => {
    e.preventDefault();

    setFilters({
      ...filters,
      searchTerm: searchTermInput,
    });
  };

  const resetFilters = useCallback((e) => {
    e.preventDefault();

    setSearchTermInput('');
    setFilters({
      pageIndex: 0,
      pageSize: 10,
      sortBy: '+applicationDate',
      status: '',
      grantOfficer: '',
      applicationId: '',
      businessCategory: '',
      businessSize: '',
      businessPremises: '',
      searchTerm: '',
    });
  }, []);

  return !error ? (
    <>
      <form onSubmit={handleSearch}>
        <TextInput
          name="searchTerm"
          label="Search"
          value={searchTermInput || ''}
          onChange={handleSearchTermChange}
        />

        <details className="govuk-details" data-module="govuk-details">
          <summary className="govuk-details__summary">
            <span className="govuk-details__summary-text">
              Advanced filters
            </span>
          </summary>
          <div className="govuk-details__text">
            <BasicSelect
              options={Object.values(APPLICATION_STATE)}
              label="Filter by Status:"
              value={filters.status}
              onChange={(status) => setValues({ status })}
            />

            {officers && (
              <BasicSelect
                options={officers}
                label="Filter by Grant Officer:"
                value={filters.grantOfficer}
                onChange={(grantOfficer) => setValues({ grantOfficer })}
              />
            )}

            <BasicSelect
              options={BUSINESS_CATEGORIES}
              label="Filter by Business Category:"
              value={filters.businessCategory}
              onChange={(businessCategory) => setValues({ businessCategory })}
            />

            <BasicSelect
              options={BUSINESS_SIZE}
              label="Filter by Business Size:"
              value={filters.businessSize}
              onChange={(businessSize) => setValues({ businessSize })}
            />

            <BasicSelect
              options={TYPE_OF_BUSINESS}
              label="Filter by Business Premises:"
              value={filters.businessPremises}
              onChange={(businessPremises) => setValues({ businessPremises })}
            />

            <BasicSelect
              options={Object.keys(DATES)}
              label="Filter by ARG round:"
              onChange={(date) => setValues({ date: DATES[date] })}
            />
          </div>
        </details>

        <div className="govuk-button-group">
          <button
            className="govuk-button"
            data-module="govuk-button"
            type="submit"
          >
            Search
          </button>
          <a
            href="#"
            className="govuk-link"
            role="button"
            onClick={resetFilters}
          >
            Clear filters
          </a>
        </div>
      </form>

      <Table
        columns={columns}
        data={data}
        fetchData={setValues}
        loading={loading}
        pageCount={pageCount}
        totalRecords={totalRecords}
        initialPage={page}
        initialPageSize={pageSize}
        initialSortBy={sort ? sort : '+applicationDate'}
      />
    </>
  ) : (
    <ErrorMessage text={error} />
  );
};

export function getQueryParametersAsObject() {
  const nextRouter = useRouter();
  const queryString = nextRouter.asPath.split(/\?/)[1];

  return Object.fromEntries(new URLSearchParams(queryString));
}

export default ApplicationsList;
