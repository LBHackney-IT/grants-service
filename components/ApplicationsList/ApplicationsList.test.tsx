import { render, waitFor, screen } from '@testing-library/react';
import ApplicationsList from './ApplicationsList';
import { fetchApplications } from '../../utils/api/applications';
jest.mock('next/router', () => ({
  push: jest.fn(),
  useRouter: () => ({
    asPath: '/',
  }),
}));
jest.mock('../../utils/api/grantOfficers', () => {
  return {
    fetchGrantOfficers: async () => {
      return {
        grantOfficers: [],
      };
    },
  };
});
jest.mock('../../utils/api/applications', () => ({
  fetchApplications: jest.fn(() =>
    Promise.resolve({ applications: [], pagination: { totalPages: 1 } })
  ),
}));

const defaultProps = {
  grantType: 'arg',
  page: 1,
  pageSize: 10,
  sort: '+applicationDate',
  status: undefined,
  grantOfficer: undefined,
  businessCategory: undefined,
  businessSize: undefined,
  businessPremises: undefined,
  applicationId: undefined,
  searchTerm: undefined,
  groups: [],
  csvDownloadGroup: undefined,
};

describe('<ApplicationsList />', () => {
  it('should render the search, filters, and results table', async () => {
    const { asFragment } = render(<ApplicationsList {...defaultProps} />);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should show a loading message while the results are pending', async () => {
    render(<ApplicationsList {...defaultProps} />);

    await waitFor(() => {
      screen.getByText('Loading...');
    });
  });

  it('should show the applications in the table once they have loaded', async () => {
    (fetchApplications as any).mockResolvedValue({
      applications: [
        {
          businessName: 'Test business name',
          clientGeneratedId: 'client-generated-id',
          applicationDate: new Date('2022-01-01').toLocaleString(),
          status: 'Unprocessed',
        },
      ],
      pagination: {
        totalPages: 1,
      },
    });
    render(<ApplicationsList {...defaultProps} />);

    await waitFor(() => {
      screen.getByText('Test business name');
    });
  });
});
