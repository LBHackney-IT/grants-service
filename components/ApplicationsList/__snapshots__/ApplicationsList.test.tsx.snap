// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationsList /> should render the search, filters, and results table 1`] = `
<DocumentFragment>
  <form>
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-label--m"
        for="searchTerm"
      >
        Search
      </label>
      <input
        class="govuk-input"
        data-testid="searchTerm"
        id="searchTerm"
        name="searchTerm"
        type="text"
        value=""
      />
    </div>
    <details
      class="govuk-details"
      data-module="govuk-details"
    >
      <summary
        class="govuk-details__summary"
      >
        <span
          class="govuk-details__summary-text"
        >
          Advanced filters
        </span>
      </summary>
      <div
        class="govuk-details__text"
      >
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by Status:"
          >
            Filter by Status:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by Status:"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="Unprocessed"
            >
              Unprocessed
            </option>
            <option
              value="Processed - Approved"
            >
              Processed - Approved
            </option>
            <option
              value="Processed - Rejected"
            >
              Processed - Rejected
            </option>
            <option
              value="Processed - Priority 2"
            >
              Processed - Priority 2
            </option>
            <option
              value="Processed - Check Bank Details"
            >
              Processed - Check Bank Details
            </option>
            <option
              value="Refer to Manager"
            >
              Refer to Manager
            </option>
            <option
              value="NFI"
            >
              NFI
            </option>
            <option
              value="Closed - Duplicate"
            >
              Closed - Duplicate
            </option>
            <option
              value="Declined - Test"
            >
              Declined - Test
            </option>
            <option
              value="Processed - Requested Information"
            >
              Processed - Requested Information
            </option>
            <option
              value="LRSG Eligible"
            >
              LRSG Eligible
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by Grant Officer:"
          >
            Filter by Grant Officer:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by Grant Officer:"
          >
            <option
              value=""
            >
              All
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by Business Category:"
          >
            Filter by Business Category:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by Business Category:"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="Section A: Agricultural, forestry and fishing"
            >
              Section A: Agricultural, forestry and fishing
            </option>
            <option
              value="Section B: Mining and quarrying"
            >
              Section B: Mining and quarrying
            </option>
            <option
              value="Section C: Manufacturing"
            >
              Section C: Manufacturing
            </option>
            <option
              value="Section D: Electricity, gas, steam and air conditioning supply"
            >
              Section D: Electricity, gas, steam and air conditioning supply
            </option>
            <option
              value="Section E: Water supply, sewerage, waste management and remediation activities"
            >
              Section E: Water supply, sewerage, waste management and remediation activities
            </option>
            <option
              value="Section F: Construction"
            >
              Section F: Construction
            </option>
            <option
              value="Section G: Wholesale and retail trade/Repair of motor vehicles and motor cycles"
            >
              Section G: Wholesale and retail trade/Repair of motor vehicles and motor cycles
            </option>
            <option
              value="Section H: Transportation and storage"
            >
              Section H: Transportation and storage
            </option>
            <option
              value="Section I: Accommodation and food services"
            >
              Section I: Accommodation and food services
            </option>
            <option
              value="Section J: Information and communication"
            >
              Section J: Information and communication
            </option>
            <option
              value="Section K: Financial and insurance activities"
            >
              Section K: Financial and insurance activities
            </option>
            <option
              value="Section L: Real estate activities"
            >
              Section L: Real estate activities
            </option>
            <option
              value="Section M: Professional,scientific and technical activities"
            >
              Section M: Professional,scientific and technical activities
            </option>
            <option
              value="Section N: Administrative and support services activities"
            >
              Section N: Administrative and support services activities
            </option>
            <option
              value="Section O: Public administration and defence/ Compulsory social security"
            >
              Section O: Public administration and defence/ Compulsory social security
            </option>
            <option
              value="Section P: Education"
            >
              Section P: Education
            </option>
            <option
              value="Section Q: Human health and social work activities"
            >
              Section Q: Human health and social work activities
            </option>
            <option
              value="Section R: Arts, entertainment and recreation"
            >
              Section R: Arts, entertainment and recreation
            </option>
            <option
              value="Section S: Other service activities"
            >
              Section S: Other service activities
            </option>
            <option
              value="Section T: Activities of households as employers/ Undifferentiated goods and services producing activities of households for own use"
            >
              Section T: Activities of households as employers/ Undifferentiated goods and services producing activities of households for own use
            </option>
            <option
              value="Section U: Activities of extraterritorial organisations and bodies"
            >
              Section U: Activities of extraterritorial organisations and bodies
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by Business Size:"
          >
            Filter by Business Size:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by Business Size:"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="Micro"
            >
              Micro
            </option>
            <option
              value="Small"
            >
              Small
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by Business Premises:"
          >
            Filter by Business Premises:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by Business Premises:"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="Individual office"
            >
              Individual office
            </option>
            <option
              value="Individual shop unit"
            >
              Individual shop unit
            </option>
            <option
              value="Market Stall"
            >
              Market Stall
            </option>
            <option
              value="Office/business unit in a shared workspace"
            >
              Office/business unit in a shared workspace
            </option>
            <option
              value="Bed and Breakfast"
            >
              Bed and Breakfast
            </option>
            <option
              value="Hotel"
            >
              Hotel
            </option>
            <option
              value="Nursery"
            >
              Nursery
            </option>
            <option
              value="Residential premises (i.e your home)"
            >
              Residential premises (i.e your home)
            </option>
            <option
              value="Other (please state):"
            >
              Other (please state):
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label"
            for="Filter by ARG round:"
          >
            Filter by ARG round:
          </label>
          <select
            class="govuk-select"
            data-testid="basic-select"
            id="Filter by ARG round:"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="Round 1"
            >
              Round 1
            </option>
            <option
              value="Round 2"
            >
              Round 2
            </option>
          </select>
        </div>
      </div>
    </details>
    <div
      class="govuk-button-group"
    >
      <button
        class="govuk-button"
        data-module="govuk-button"
        type="submit"
      >
        Search
      </button>
      <a
        class="govuk-link"
        role="button"
      >
        Clear filters
      </a>
    </div>
  </form>
  <table
    class="govuk-table"
    role="table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
        role="row"
      >
        <th
          class="govuk-table__header"
          colspan="1"
          role="columnheader"
          scope="col"
        >
          <span
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Business Name
          </span>
        </th>
        <th
          class="govuk-table__header"
          colspan="1"
          role="columnheader"
          scope="col"
        >
          <span>
            Application ID
          </span>
        </th>
        <th
          class="govuk-table__header"
          colspan="1"
          role="columnheader"
          scope="col"
        >
          <span
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Submission ↑
          </span>
        </th>
        <th
          class="govuk-table__header"
          colspan="1"
          role="columnheader"
          scope="col"
        >
          <span>
            Status
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
      role="rowgroup"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
          colspan="10000"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="pagination"
  >
    <button
      disabled=""
    >
      &lt;&lt;
    </button>
     
    <button
      disabled=""
    >
      &lt;
    </button>
     
    <button
      disabled=""
    >
      &gt;
    </button>
     
    <button
      disabled=""
    >
      &gt;&gt;
    </button>
     
    <span>
      Page 
      <strong>
        1 of 0
      </strong>
       
    </span>
    <span>
      | Go to page: 
      <input
        style="width: 100px;"
        type="number"
        value="1"
      />
    </span>
     
    <select>
      <option
        value="10"
      >
        Show 10
      </option>
      <option
        value="20"
      >
        Show 20
      </option>
      <option
        value="30"
      >
        Show 30
      </option>
      <option
        value="40"
      >
        Show 40
      </option>
      <option
        value="50"
      >
        Show 50
      </option>
    </select>
  </div>
  <hr
    class="govuk-section-break govuk-section-break--l govuk-section-break--visible"
  />
  <p>
    <a
      href="/api/csv/applications?grantType=arg"
      target="_blank"
    >
      Download Applications CSV
    </a>
  </p>
  <h2>
    Export payment details
  </h2>
  <div>
    <p>
      These downloads are disabled as you are not part of the 'undefined' user group
    </p>
    <div>
      <button
        class="govuk-button govuk-button--secondary govuk-!-margin-right-1"
        data-module="govuk-button"
        disabled=""
        round="1"
      >
        Export Panel Approved Payments (Round 1)
      </button>
    </div>
    <div>
      <button
        class="govuk-button govuk-button--secondary govuk-!-margin-right-1"
        data-module="govuk-button"
        disabled=""
        round="2"
      >
        Export Panel Approved Payments (Round 2)
      </button>
    </div>
  </div>
</DocumentFragment>
`;
